### PR TITLE
[SCFD-3113] Added validation that vector should not contain None components

### PR DESCRIPTION
**Pull request by benflexcompute**

### CODE REVIEW

File: flow360/component/simulation/unit_system.py [hunk 1]
```diff
@@ -157,25 +157,39 @@ def _has_dimensions(quant, dim):
...
+            dimensioned_value = unyt_type(value["value"], value["units"], dtype=np.float64)
+            # Check for Nan. This usually is a result of None being in the input value.
+            # with dtype=np.float64, None will be converted to np.nan.
+            # We need (IIRC) np.float64 to ensure consistent hashing
+            if kwargs.get("allow_inf_nan", False) is False:
+                if np.ndim(dimensioned_value.value) == 0 and np.isnan(dimensioned_value.value):
+                    raise ValueError("NaN or None found in input scalar which is not allowed.")
+                if np.ndim(dimensioned_value.value) > 0 and any(np.isnan(dimensioned_value.value)):
+                    raise ValueError("NaN or None found in input array which is not allowed.")
+            return dimensioned_value
```
Comment (maciej-flexcompute) [inline flow360/component/simulation/unit_system.py:181]: can we put this logic to a separate validator and keep this validator only a parser? See the logic of `_VectorType.validate()` it checks step by step all conditions, it is easier to follow.

Comment (benflexcompute) [reply flow360/component/simulation/unit_system.py]: Done

File: flow360/component/simulation/unit_system.py [hunk 2]
```diff
@@ -358,12 +367,18 @@ def get_class_object(cls, dim_type, **kwargs):
...
+                        **{k: v for k, v in kwargs.items() if k != "allow_inf_nan"}
+                    ),
+                ]
 
-            def validate(con_cls, value, *args):
+            def validate(con_cls, value, *args, **kwargs):
                 """Additional validator for value"""
                 try:
-                    dimensioned_value = dim_type.validate(value)
+                    dimensioned_value = dim_type.validate(value, kwargs)
```
Comment (maciej-flexcompute) [inline flow360/component/simulation/unit_system.py:381]: shouldn't this be:
```suggestion
                    dimensioned_value = dim_type.validate(value, **kwargs)
```
?

Comment (benflexcompute) [reply flow360/component/simulation/unit_system.py:381]: You are correct.

**Review by maciej-flexcompute — APPROVED** [review]

**Pull request merged**